src/cargo/core/compiler/mod.rs

```diff
@@ -798,6 +798,9 @@ fn add_error_format_and_color(cx: &Context<'_, '_>, cmd: &mut ProcessBuilder, pi
         // to emit a message that cargo will intercept.
         json.push_str(",artifacts");
     }
+    // Always emit a future-incompat report, so we can report
+    // future-incompat dependencies to the user
+    json.push_str(",future-incompat");
```

Comment:
I believe this still needs the feature detection until it lands on stable. CI tests cargo with stable rustc in addition to nightly rustc. https://github.com/rust-lang/cargo/blob/0507d9f5cec689242427599cc1e84c39c62b59c4/.github/workflows/main.yml#L37-L54

Reply:
Yea, I think this will need to probe that `rustc` supports the `future-incompat` option. That would go in [`TargetInfo`](https://github.com/rust-lang/cargo/blob/f9089fc5be17e9ddfa0bcf79999572fe7e247cb1/src/cargo/core/compiler/build_context/target_info.rs#L22).  I think it would look essentially the same as `supports_split_debuginfo`.

Reply:
I've added a prove for `--json future-incompat`